web/src/sidebar_ui.js

```diff
@@ -22,30 +28,63 @@ export function hide_streamlist_sidebar() {
     $(".app-main .column-left").removeClass("expanded");
 }
 
-$("body").on("click", ".login_button", (e) => {
-    e.preventDefault();
-    e.stopPropagation();
-    window.location.href = hash_util.build_login_link();
-});
+export function register_click_handlers() {
+    document.addEventListener(
+        "click",
+        (e) => {
+            if (window.innerWidth > media_breakpoints_num.xl) {
+                // Sidebars are always visible beyond xl breakpoint.
+                return;
+            }
 
-$("#userlist-toggle-button").on("click", (e) => {
-    e.preventDefault();
-    e.stopPropagation();
+            const $left_column = $(".app-main .column-left");
+            const $right_column = $(".app-main .column-right");
+            const $elt = $(e.target);
+            const toggle_left_sidebar = Boolean($elt.closest("#streamlist-toggle-button").length);
+            const toggle_right_sidebar = Boolean($elt.closest("#userlist-toggle-button").length);
+            const left_sidebar_visible = $left_column.hasClass("expanded");
+            const right_sidebar_visible = $right_column.hasClass("expanded");
```

Comment:
It might be good to have another early return check of the form `if (!left_sidebar_visible && !right_sidebar_visible) { return ; }`

Reply:
but, this function is also used to show the sidebars so, that would be wrong.